### PR TITLE
LayerIndex resolver for image component

### PR DESCRIPTION
**Pull request by md2119**
## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- ~Evaluated and added CHANGELOG entry if required~
- ~Determined and documented upgrade steps~
- ~Documented user facing changes~
 
If any of these don't apply, please comment below.

## Testing Performed
CI

### CODE REVIEW

File: central/graphql/resolvers/components_v1.go [hunk 1]
```diff
@@ -113,13 +113,13 @@ func (eicr *EmbeddedImageScanComponentResolver) RiskScore(ctx context.Context) f
 }
 
 // LayerIndex is the index in the parent image.
-func (eicr *EmbeddedImageScanComponentResolver) LayerIndex() *int32 {
+func (eicr *EmbeddedImageScanComponentResolver) LayerIndex() (*int32, error) {
 	w, ok := eicr.data.GetHasLayerIndex().(*storage.EmbeddedImageScanComponent_LayerIndex)
 	if !ok {
-		return nil
+		return nil, nil
```
[the remark this answers is not in the document]
Comment (md2119) [reply central/graphql/resolvers/components_v1.go:119]: We should not return an error. This is merely to make the new and the old interface consistent.

File: central/graphql/resolvers/image_components.go [hunk 2]
```diff
@@ -411,8 +411,20 @@ func (resolver *imageComponentResolver) ID(_ context.Context) graphql.ID {
...
-func (resolver *imageComponentResolver) LayerIndex() *int32 {
-	return nil
+func (resolver *imageComponentResolver) LayerIndex() (*int32, error) {
+	scope, hasScope := scoped.GetScope(resolver.ctx)
+	if !hasScope || scope.Level != v1.SearchCategory_IMAGES {
+		return nil, nil
+	}
+	edges, err := resolver.root.ImageComponentEdgeDataStore.SearchRawEdges(resolver.ctx, resolver.componentQuery())
+	if err != nil {
+		return nil, nil
```
[the remark this answers is not in the document]
Comment (md2119) [reply central/graphql/resolvers/image_components.go:421]: 😱 
good catch

File: central/graphql/resolvers/image_components.go [hunk 3]
```diff
@@ -68,7 +68,7 @@ type ImageComponentResolver interface {
 	ImageVulnerabilityCounter(ctx context.Context, args RawQuery) (*VulnerabilityCounterResolver, error)
 	ImageVulnerabilities(ctx context.Context, args PaginatedQuery) ([]ImageVulnerabilityResolver, error)
 	LastScanned(ctx context.Context) (*graphql.Time, error)
-	LayerIndex() *int32
+	LayerIndex() (*int32, error)
```
[the remark this answers is not in the document]
Comment (md2119) [reply central/graphql/resolvers/image_components.go:71]: Actually, there is no need to return the pointer. Changing the interface to return int32.

**Review by charmik-redhat — APPROVED** [review]

File: central/graphql/resolvers/image_components.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (md2119) [reply central/graphql/resolvers/image_components.go]: Apparently, we need the pointer. `null` to indicate the layer index is not set whereas `0` indicates a index 0.

**Pull request merged**